HIP/hip-1046.md

```diff
@@ -2,147 +2,219 @@
+Book. To work around this gRPC-Web endpoints are hard-coded in the JavaScript
+SDK. This is not a scalable or efficient solution as it requires manual updates
+and coordination. The problem will only get worse as Hedera continues
+decentralization of the network and independent node operation, where this
+coordination is not possible.
+
+We propose enhancing the Hedera address book to include gRPC-Web endpoint
+information, allowing the JavaScript SDK to dynamically determine the
+appropriate endpoints. This will be achieved by:
+
+1. **Extending the Address Book Schema**: Modify the address book schema to
+   include entries for gRPC-Web endpoints.
+2. **Dynamic Discovery**: Update the JavaScript SDK to dynamically retrieve the
+   gRPC-Web endpoints from the address book.
+3. **Query Availability**: Update the Mirror Node to recognize gRPC-Web entries
+   in the address book and make those entries available for the SDK to query.
+
+By implementing these changes, we can ensure that frontend applications can
+securely and efficiently connect to the Hedera network using dynamically
+discovered, decentralized TLS endpoints. This will improve the network's
+resilience, scalability, and overall user experience.
 
 ## Rationale
 
-Given that this proposal involves an incremental change to the already existing Hedera Address Book, the design is straightforward. The primary goal is to enhance the address book to include gRPC-Web endpoint information, allowing frontend applications to dynamically determine the appropriate endpoints. 
+Given that this proposal involves an incremental change to the already existing
+Hedera Address Book, the design is straightforward. The primary goal is to
+enhance the address book to include gRPC-Web endpoint information, allowing
+frontend applications to dynamically determine the appropriate endpoints.
 
-*Why:* Updating the JavaScript SDK to dynamically retrieve gRPC-Web endpoints from the address book eliminates the need for hard-coded endpoints. This design decision enhances flexibility and scalability, allowing frontend applications to adapt to changes in the network configuration without requiring manual updates.
+*Why:* Updating the JavaScript SDK to dynamically retrieve gRPC-Web endpoints
+from the address book eliminates the need for hard-coded endpoints. This design
+decision enhances flexibility and scalability, allowing frontend applications to
+adapt to changes in the network configuration without requiring manual updates.
 
-*Alternatives Considered:* Another alternative was to continue with the current practice of hard-coding endpoints within the SDK. However, this method is not sustainable as it requires frequent updates and coordination, particularly as Hedera continues to decentralize. Dynamic discovery provides a more robust and future-proof solution.
+*Alternatives Considered:* Another alternative was to continue with the current
```

Comment:
This 80 character limit seems too restrictive and conflicts with our Java coding [standards](https://www.notion.so/swirldslabs/007-Java-Code-Style-a0130a7059b044f68447b9ca3a5242b3#9bc7be511ea94b588d407491c28068c3) to use 120. The standards just mention java so not sure about other formats. Is it documented somewhere that we should use 80 for markdown?

Reply:
There is no standard for markdown that I know of, but github enforces 80 characters in much of the review interface (particularly suggestions), so I habitually reformat HIP markdown to 80 characters for ease of review.
I do not think there is any requirement to do so, however.